core/rfb.js

```diff
@@ -1271,6 +1283,127 @@ export default class RFB extends EventTargetMixin {
         return this._fail("No supported sub-auth types!");
     }
 
+    _negotiate_ra2ne_auth() {
+        if (this._rfb_ra2_state.state === 0) {
+            // Send client RSA public key
+            if (!this._rfb_ra2_state.has_set_client_rsa_key()) {
+                this.dispatchEvent(new CustomEvent(
+                    "rsakeyget",
+                    { detail: {} }));
+                return false;
+            }
+            this._sock.send(this._rfb_ra2_state.client_public_key);
+            this._rfb_ra2_state.state = 1;
+            return false;
+        } else if (this._rfb_ra2_state.state === 1) {
+            // Receive server RSA public key
+            // Send RSA encrypted client random
+            if (this._sock.rQwait("ra2 public key", 516, 0)) {
+                return false;
+            }
+            if (!this._rfb_ra2_state.parse_server_rsa_key(this._sock.rQshiftBytes(516))) {
+                return this._fail("Invalid RSA key");
+            }
```

Comment:
We need to let the user approve this key as well, so we can be sure we are talking to the right server.

I guess we need a new event for this. Or if we add a field to the credentials where approved server keys can be added.